pkg/controllers/job/job_controller_actions.go

```diff
@@ -69,7 +69,7 @@ func (cc *jobcontroller) killJob(jobInfo *apis.JobInfo, podRetainPhase state.Pha
 
 			maxRetry := job.Spec.MaxRetry
 			lastRetry := false
-			if job.Status.RetryCount >= maxRetry-1 {
+			if job.Status.RetryCount > maxRetry-1 {
```

Comment:
on the last retry what's the retry count? my naive reading is that it would make this condition always be false. i.e. on the 2nd retry with maxRetry=3 you get `2 > 3-1` so lastRetry is never true. 

I don't quite understand how this breaks for `maxRetry == 1` but works for `maxRetry > 1`.

Reply:
As what I tested, the retry count is 0 when the job starts. So as to the example you list, on the 2nd retry with maxRetry=3, it is not the last retry but there is another chance to retry. That's the last retry.